components/brave_vpn/browser/connection/brave_vpn_os_connection_api.h

```diff
@@ -42,26 +47,65 @@ class BraveVPNOSConnectionAPI {
     ~Observer() override = default;
   };
 
-  virtual mojom::ConnectionState GetConnectionState() const = 0;
-  virtual void ResetConnectionState() = 0;
+  // Shared APIs implementation between IKEv2/Wireguard connections.
+  mojom::ConnectionState GetConnectionState() const;
+  void AddObserver(Observer* observer);
+  void RemoveObserver(Observer* observer);
+  void SetConnectionState(mojom::ConnectionState state);
```

Comment:
I found that this method is only used in test code.
How about renaming to `SetConnectionStateForTesting()`  and moving to `private`?

Reply:
done